src/writer.js

```diff
@@ -71,6 +68,14 @@ class Writer {
       }
     }
 
+    if (this._url.protocol === 'unix:') {
+      options.socketPath = this._url.pathname
+    } else {
+      options.protocol = this._url.protocol
```

Comment:
Why not set the protocol regardless of whether it's `http:` or `unix:`?

Reply:
Setting `options.protocol = 'unix;'` hits [this condition](https://github.com/nodejs/node/blob/3c277d7e92adf1464e41d27677fcdc05fec41a2b/lib/_http_client.js#L118-L132) with an error: `Protocol "unix:" not supported. Expected "http:"`. 

It looks like we could potentially swap out the agent with one that expects the unix protocol. I've never looked at the stdlib this closely. What do you think?

Reply:
Thanks for the clarification! If it works without the protocol, I'm fine to discard it.

Reply:
:+1: thanks for the quick review!